### PR TITLE
Refactor catalog UI

### PR DESCRIPTION
**Pull request by akucharska**
**Description**

Changes proposed in this pull request:

- Refactor Catalog UI to make it testable and easier to connect with websockets.

**Related issue(s)**

https://github.com/kyma-project/console/issues/1340

### CODE REVIEW

File: service-catalog-ui/catalog/src/components/App/App.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+Modal.MODAL_APP_REF = '#root';
+
+const ServiceDetails = ({ match }) => (
+  <ServiceClassDetails name={match.params.name} />
+);
+
+export default function App() {
+  return (
+    <div className="ph3 pv1 background-gray">
```
Comment (Wawrzyn321) [inline service-catalog-ui/catalog/src/components/App/App.js]: It looks like we can remove this div.

File: service-catalog-ui/catalog/src/components/ServiceClassDetails/CreateInstanceModal/CreateInstanceModal.scss [hunk 2]
```diff
@@ -0,0 +1,14 @@
+.grid-wrapper {
+    display: grid;
+    grid-template-columns: repeat(auto-fill, minmax(300px, 1fr) ) ;
+    grid-gap: 16px;
+}
+
+.separator {
```
Comment (Wawrzyn321) [inline service-catalog-ui/catalog/src/components/ServiceClassDetails/CreateInstanceModal/CreateInstanceModal.scss]: We should use less generic names for CSS classes.

File: service-catalog-ui/catalog/src/components/ServiceClassDetails/CreateInstanceModal/test/CreateInstanceModal.test.js [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+    expect(
+      component
+        .find('form#createInstanceForm')
+        .instance()
+        .checkValidity(),
+    ).toEqual(false);
+  });
+
+  it('Creates instance properly', async () => {
+    const ref = React.createRef();
```
Comment (Wawrzyn321) [inline service-catalog-ui/catalog/src/components/ServiceClassDetails/CreateInstanceModal/test/CreateInstanceModal.test.js]: { current: null } would do. Like in `CreateInstanceModal::Shows error notification if an error occured`.

File: service-catalog-ui/catalog/src/components/ServiceClassDetails/ServiceClassDetails.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+          id="add-instance-modal"
+          item={serviceClass}
+          renderForm={props => <CreateInstanceModal {...props} />}
+        />
+      </ServiceClassDetailsHeader>
+
+      <ServiceClassDetailsWrapper phoneRows>
+        {backendModuleExists('cms') && backendModuleExists('assetstore') ? (
+          <ServiceClassTabs serviceClass={serviceClass} />
+        ) : null}
```
Comment (Wawrzyn321) [inline service-catalog-ui/catalog/src/components/ServiceClassDetails/ServiceClassDetails.js]: &&

File: service-catalog-ui/catalog/src/components/ServiceClassDetails/test/ServiceClassDetails.test.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+      };
+    },
+    uxManager: function() {
+      return {
+        addBackdrop: mockAddBackdrop,
+        removeBackdrop: mockRemoveBackdrop,
+      };
+    },
+  };
+});
```
Comment (Wawrzyn321) [inline service-catalog-ui/catalog/src/components/ServiceClassDetails/test/ServiceClassDetails.test.js]: Either stick to `function` syntax, or (maybe better) use arrow functions.

File: service-catalog-ui/catalog/src/components/ServiceClassList/ServiceClassToolbar/ServiceClassToolbar.component.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+import { Toolbar } from '@kyma-project/react-components';
+
+import SearchDropdown from './SearchDropdown.component';
+
+const ServiceClassToolbar = ({ searchFn, serviceClassesExists }) => {
+  return (
+    <Toolbar background="#fff" title={serviceClassConstants.title}>
+      {serviceClassesExists ? (
+        <SearchDropdown onChange={e => searchFn(e.target.value)} />
+      ) : null}
```
Comment (Wawrzyn321) [inline service-catalog-ui/catalog/src/components/ServiceClassList/ServiceClassToolbar/ServiceClassToolbar.component.js]: &&

File: service-catalog-ui/catalog/src/shared/ModalWithForm/ModalWithForm.js [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+      isFormValid(formElementRef, reportValidity) &&
+      isJsonSchemaFormValid(jsonSchemaFormRef);
+
+    if (isValid !== _isEveryFormValid) {
+      setValid(_isEveryFormValid);
+    }
+  }
+
+  useEffect(() => {
+    setTimeout(() => checkAllForms(true));
```
Comment (Wawrzyn321) [inline service-catalog-ui/catalog/src/shared/ModalWithForm/ModalWithForm.js]:
```suggestion
    setImmediate(() => checkAllForms(true));
```
I suppose.

File: service-catalog-ui/catalog/src/shared/ModalWithForm/ModalWithForm.js [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+      if (e.target.getAttribute('data-ignore-visual-validation')) {
+        return;
+      }
+
+      // current element validity
+      if (e.target.checkValidity()) {
+        e.target.classList.remove('is-invalid');
+      } else {
+        e.target.classList.add('is-invalid');
+      }
```
Comment (Wawrzyn321) [inline service-catalog-ui/catalog/src/shared/ModalWithForm/ModalWithForm.js]: You could use 
```
const isValid = e.target.checkValidity();
e.target.classList.toggle('is-invalid', !isValid);
```

File: service-catalog-ui/catalog/src/shared/ModalWithForm/ModalWithForm.js (hunk 7 above)
Comment (akucharska) [reply service-catalog-ui/catalog/src/shared/ModalWithForm/ModalWithForm.js]: This change brakes the tests

**Comment by akucharska** [comment]
Merged in https://github.com/kyma-project/console/pull/1402

**Pull request closed without merging**